### PR TITLE
Use `this.ctx.storage` to store session data

### PR DESCRIPTION
**Pull request by threepointone**
(instead of connection.state)

### CODE REVIEW

**Comment by third774** [comment]
Hope you don't mind — I threw up a commit that I _think_ gets this into working order the way we want. Now that room state is derived from the session storage we can't rely on `connection.send()` throwing errors to remove users from the room state.

The other change I made was that I noticed we were calling `JSON.stringify` in some cases, but not in others when setting the user state.

```ts
this.ctx.storage.put(`session-${connection.id}`, JSON.stringify(user))
```

I went ahead and removed this since DO storage accepts structured clone compatible data types.

File: app/durableObjects/ChatRoom.server.ts [hunk 1]
```diff
@@ -172,14 +169,13 @@ export class ChatRoom extends Server<Env> {
 							const otherUser = await this.ctx.storage.get<User>(
 								`session-${data.id}`
 							)
-							this.ctx.storage.put(`session-${data.id}`, {
+							await this.ctx.storage.put(`session-${data.id}`, {
```
Comment (threepointone) [inline app/durableObjects/ChatRoom.server.ts:172]: good catch, I really need to land the new linting rules, I'll do so next week

File: app/durableObjects/ChatRoom.server.ts [hunk 2]
```diff
@@ -237,9 +233,19 @@ export class ChatRoom extends Server<Env> {
...
 	}
 
+	async cleanupOldConnections() {
+		const connections = [...this.getConnections()]
+		const sessionsToCleanUp = await this.ctx.storage.list<User>()
+		connections.forEach((connection) =>
+			sessionsToCleanUp.delete(`session-${connection.id}`)
+		)
+		return Promise.all(
+			[...sessionsToCleanUp.keys()].map((key) => this.ctx.storage.delete(key))
```
Comment (threepointone) [inline app/durableObjects/ChatRoom.server.ts:243]: you can pass multiple keys to .delete

**Comment by threepointone** [comment]
looks good! feel free to review and land this whenever you feel like it

File: app/durableObjects/ChatRoom.server.ts (hunk 2 above)
Comment (third774) [reply app/durableObjects/ChatRoom.server.ts]: Ah, nice — fixed in force push.

**Review by third774 — APPROVED** [review]

**Pull request merged**